src/mongoc/mongoc-server-description.h

```diff
@@ -22,6 +22,19 @@
 #include "mongoc-read-prefs.h"
 #include "mongoc-host-list.h"
 
+typedef enum
+{
+   MONGOC_SERVER_UNKNOWN,
+   MONGOC_SERVER_STANDALONE,
+   MONGOC_SERVER_MONGOS,
+   MONGOC_SERVER_POSSIBLE_PRIMARY,
+   MONGOC_SERVER_RS_PRIMARY,
+   MONGOC_SERVER_RS_SECONDARY,
+   MONGOC_SERVER_RS_ARBITER,
+   MONGOC_SERVER_RS_OTHER,
+   MONGOC_SERVER_RS_GHOST,
```

Comment:
Because the MONGOC_SERVER_DESCRIPTION_TYPES definition is decoupled from the enum, you probably should add at least a comment here that that value needs to be updated if the enum is extended. I would probably rather have the MONGOC_SERVER_DESCRIPTION_TYPES here to prevent them from going out of sync.